src/drivers/apa102.rs

```diff
@@ -3,56 +3,69 @@ use smart_leds::RGB8;
 
 use crate::drivers::LedDriver;
 
-pub struct Apa102<SPI: SpiBus<u8>> {
+/// APA102 LED Driver supporting arbitrary number of LEDs.
+///
+/// The caller is responsible for providing a buffer of appropriate size.
```

Comment:
would be better to explain how large the buffer is supposed to be. You seem to have a formula, you could document it so that the caller doesn't have to guess

Reply:
(this applies to all drivers)

Reply:
make sense